src/joystick/SDL_joystick.c

```diff
@@ -969,14 +969,16 @@ static void AttemptSensorFusion(SDL_Joystick *joystick, bool invert_sensors)
 
             if (!joystick->accel_sensor && SDL_GetSensorTypeForID(sensor) == SDL_SENSOR_ACCEL) {
                 // Increment the sensor subsystem reference count
-                SDL_InitSubSystem(SDL_INIT_SENSOR);
+                const int rc = SDL_InitSubSystem(SDL_INIT_SENSOR);
+                SDL_assert(rc);
```

Comment:
Adding an assert inside SDL is not the correct approach, I think.

Reply:
What would you suggest here instead? I agree that an assertion isn't ideal, but it's the closest thing to what this code was doing originally as there was an implicit assumption that `SDL_InitSubSystem(SDL_INIT_SENSOR)` would pass.